crates/fuel-core/src/database.rs

```diff
@@ -227,6 +227,13 @@ where
         );
         GenesisDatabase::new(self.into_inner().data)
     }
+
+    /// !!!! WARNING !!!!
+    /// This method is unsafe because it doesn't check if the height is already set.
+    /// This allows overriding the genesis state with a new one.
+    pub fn unsafe_into_genesis(self) -> GenesisDatabase<Description> {
+        GenesisDatabase::new(self.into_inner().data)
+    }
```

Comment:
We need to add a todo to rework this. It breaks all procession that we have for database right now.

Reply:
already done #2329

Reply:
I meant in the code itself